distributed/client.py

```diff
@@ -2342,24 +2344,28 @@ def list_datasets(self, **kwargs):
         """
         return self.sync(self.scheduler.publish_list, **kwargs)
 
-    async def _get_dataset(self, name):
+    async def _get_dataset(self, name, default):
         with self.as_current():
             out = await self.scheduler.publish_get(name=name, client=self.id)
 
         if out is None:
-            raise KeyError(f"Dataset '{name}' not found")
+            if default is None:
```

Comment:
What if we want a default of `None`? Perhaps use use placeholder like `"_no_default_"`.